kernel/sched.c

```diff
@@ -561,9 +561,13 @@ void _priq_dumb_remove(sys_dlist_t *pq, struct k_thread *thread)
 
 struct k_thread *_priq_dumb_best(sys_dlist_t *pq)
 {
+	struct k_thread *t = NULL;
```

Comment:
The fix is correct, but can you clarity the explanation?  It was a bug, call it a bug, e.g:

    kernel/sched: Fix empty list detection

    CONTAINER_OF() on a NULL pointer returns some offset around NULL and
    not another NULL pointer.  We have to check for that ourselves.

    This only worked because the dnode happened to be at the start of the struct.